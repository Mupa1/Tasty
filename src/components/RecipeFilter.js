import React from 'react';
import PropTypes from 'prop-types';

const RecipeFilter = ({ updateFilter, filter }) => {
  const categories = [
    'Chicken',
    'Beef',
    'Pizza',
    'Pasta',
    'Fish',
  ];

  const handleFilterChange = e => {
    updateFilter(e.target.value);
  };

  return (
    <div>
      <select onChange={handleFilterChange} name="category" defaultValue={filter}>
        {categories.map(category => (
          <option key={category}>{category}</option>
        ))}
      </select>
    </div>
  );
};

RecipeFilter.propTypes = {
  updateFilter: PropTypes.func.isRequired,
  filter: PropTypes.string.isRequired,
};

export default RecipeFilter;
