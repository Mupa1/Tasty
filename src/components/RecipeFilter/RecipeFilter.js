import React from 'react';
import PropTypes from 'prop-types';

import './RecipeFilter.module.css';

const RecipeFilter = ({ updateFilter, filter }) => {
  const categories = [
    'Chicken',
    'Beef',
    'Pizza',
    'Pasta',
    'Fish',
  ];

  const handleFilterChange = e => {
    updateFilter(e.target.value);
  };

  return (
    <select onChange={handleFilterChange} name="category" defaultValue={filter}>
      {categories.map(category => (
        <option key={category}>{category}</option>
      ))}
    </select>
  );
};

RecipeFilter.propTypes = {
  updateFilter: PropTypes.func.isRequired,
  filter: PropTypes.string.isRequired,
};

export default RecipeFilter;
