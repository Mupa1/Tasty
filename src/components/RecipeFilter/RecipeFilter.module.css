select {
  border: solid 1px #28a745;
  width: 300px;
  height: 30px;
  margin: 40px auto 30px auto;
  font-size: 18px;
}

select:focus {
  outline-color: #28a745;
}
