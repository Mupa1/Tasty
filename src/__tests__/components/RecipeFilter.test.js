import React from 'react';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

import RecipeFilter from '../../components/RecipeFilter/RecipeFilter';

afterEach(cleanup);

it('matches RecipeFilter snapshot', () => {
  const tree = renderer
    .create(<RecipeFilter updateFilter={() => 'mock'} filter="Pizza" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders a select input', () => {
  const { queryAllByTestId } = render(
    <RecipeFilter updateFilter={() => 'mock'} filter="Pizza" />,
  );
  const result = queryAllByTestId('categories');
  expect(result).toBeTruthy();
  expect(result[0]).toHaveTextContent('Pizza');
});
