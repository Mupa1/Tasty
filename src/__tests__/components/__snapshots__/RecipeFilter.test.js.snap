// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches RecipeFilter snapshot 1`] = `
<select
  data-testid="categories"
  defaultValue="Pizza"
  name="category"
  onChange={[Function]}
>
  <option>
    Chicken
  </option>
  <option>
    Beef
  </option>
  <option>
    Pizza
  </option>
  <option>
    Pasta
  </option>
  <option>
    Fish
  </option>
</select>
`;
